axum-core/src/ext_traits/request.rs

```diff
@@ -259,12 +258,18 @@ pub trait RequestExt: sealed::Sealed + Sized {
     /// Apply the [default body limit](crate::extract::DefaultBodyLimit).
     ///
     /// If it is disabled, return the request as-is in `Err`.
-    fn with_limited_body(self) -> Result<Request<Limited<Body>>, Request>;
+    ///
+    /// Note that while the `Ok` and `Err` types are the same, [`http_body_util::Limited`] will have
+    /// been applied in the `Ok` case.
+    fn with_limited_body(self) -> Result<Request, Request>;
```

Comment:
I think we should return `Result` now that the types are the same anyways. Same for `into_limited_body`. If we want people to be able to detect whether this would do anything, that could be a separate method.